StayDetailView 기본 동작 구현

kjoonk opened the pull request:
상세 뷰에 대한 기본 구현이에요.
- TabBarController > NavigationController > StayListViewController 구조로 바꿨어요.
- Cell 터치 시 상세뷰로 이동: collectionView의 delegate를 썼는데, 셀 크기가 작아지는 문제가 있네요. 같이 해결해보거나, 다른 방법을 써야 할 것 같아요.
- DetailSectionView: 상세 뷰를 이런 식으로 나누면 될 것 같아요. 테스트용으로 label, button만 두었어요.
- RoundedBorder: 공통 스타일을 이런 식으로 적용하면 어떨까 싶어요. 프로토콜이랑 익스텐션 써서요. 기본값을 어떻게 처리할까가 고민이긴 한데, 메소드의 인자로 우선 구현했어요. 다른 방법으로는 상속받아서 사용하는 방법이 있겠고요. 
- DetailButton: RoundedBorder를 적용한 예시에요. DetailSectionView 안에 있는 버튼이에요.

kjoonk committed:
[iOS] feat: StayDetailViewController 생성

-  대응하는 xib 생성

issue: #53

kjoonk committed:
[iOS] style: 오타 수정

- seperator -> separator

kjoonk committed:
[iOS] feat: StayList에서 Cell 선택하면 상세화면으로 전환

- StayList의 CollectionView delegate로 구현
- FIXME: delegate로 지정하면 cell 크기가 바뀌는 문제

issue: #53

kjoonk committed:
[iOS] feat: 둥근 외곽선에 대한 protocol 구현
- RoundedBorder protocol
- 기본값은 extension으로 구현

issue: #53

kjoonk committed:
[iOS] feat: StayListViewController를 NavigationController에 포함함

- TabBarController > NavigationController > StayListViewController

issue: #53

corykim0829 commented on iOS/AirbnbApp/AirbnbApp/StayList/Controllers/StayListViewController.swift:
```diff
@@ -59,13 +65,16 @@ class StayListViewController: UIViewController {
     private func configureCollectionView() {
         stayListCollectionView.dataSource = stayListCollectionViewDataSource
+
+        // FIXME: delegate 지정하면 셀 크기가 이상하게 바뀜
+        stayListCollectionView.delegate = stayListCollectionViewDelegate
```
이 부분은 stayListCollectionView 클래스 파일에 가면, 제가 그 안에서 delegate를 지정해놨는데, 여기서 delegate를 변경해서 그럴거에요.

kjoonk replied:
음.. 코리 말대로 delegate 관련해서 찾아보니 `UICollectionView`와 `UICollectionViewFlowLayout` 두 개가 같은 `delegate`를 쓰네요.  [문서](https://developer.apple.com/documentation/uikit/uicollectionviewflowlayout#overview)
두 delegate 구현부를 하나의 클래스로 옮겼더니 잘 작동해요.

corykim0829 commented on iOS/AirbnbApp/AirbnbApp/TabBar/TabBarController.swift:
```diff
@@ -2,7 +2,7 @@ import UIKit
 class TabBarController: UITabBarController {
     
-    private var stayListViewController: StayListViewController!
+    private var navViewController: UINavigationController!
```
큰 차이는 없지만 stayListViewController를 내부 프로퍼티로 살리는건 어떨까요?

kjoonk replied:
`TabBarController`의 속성으로 살리자는 건가요? 지워도 괜찮길래 지웠어요. 나중에 `stayListViewController`를 직접 제어할 일이 있으면 그때 살려도 되지 않을까 싶어요.

kjoonk replied:
`navViewController` 속성도 지워버렸어요. 어차피 `viewControllers` 속성으로 가지고 있어서요.

corykim0829 reviewed (changes requested):
수고하셨습니다!

kjoonk committed:
[iOS] fix: CollectionView의 Cell 크기가 적용되지 않던 문제 수정

- CollectionView와 CollectionViewFlowLayout의 delegate 관련 문제였음
- 이 둘이 같은 delegate를 사용하여 FlowLayout 쪽이 적용되지 않았음

issue: #56

kjoonk committed:
[iOS] refactor: NavigationController 생성 부분 변경

- 별도 메소드로 분리
- 뷰컨트롤러를 반환 받아 처리
- 속성 제거

issue: #53

corykim0829 reviewed (approved):
수고하셨습니다!!
